### ISSUE TITLE
An unhandled panic occurred in the Encore compiler: runtime error: index out of range [3] with length 3

### ISSUE THREAD

**Issue by saujanreshmi** [encoredev/encore#754]
restoring build cache
no build cache found, skipping
── Internal compiler error ────────────────────────────────────────────[E0001]──

An unhandled panic occurred in the Encore compiler: runtime error: index out of
range [3] with length 3

This is a bug in Encore and should not have occurred. Please report this issue
to the Encore team either on Github at
https://github.com/encoredev/encore/issues/new and include this error.

╭─[Stack Trace]
├─▶ v2builder.BuilderImpl.Parse.func1.1 /encore-compiler/v2/v2builder/v2builder.go:47
├─▶ app.(*Desc).validateAPIs            /encore-compiler/v2/app/validate_apis.go:76
├─▶ app.(*Desc).validate                /encore-compiler/v2/app/validate.go:21
├─▶ app.ValidateAndDescribe             /encore-compiler/v2/app/app.go:104
├─▶ v2builder.BuilderImpl.Parse.func1   /encore-compiler/v2/v2builder/v2builder.go:84
├─▶ etrace.Sync2[...]                   /encore-compiler/internal/etrace/etrace.go:20
╰─[... remaining frames omitted ...]

**Comment by eandre** [comment]
Thanks for the report @saujanreshmi! From the stack trace, it seems like the issue is most likely due to having defined path parameters in the `path` annotation, but not having a corresponding parameter on the function signature. 

A correct signature might be:

```
//encore:api public method=POST path=/blog/:postID/comment
func AddComment(ctx context.Context, postID int, params *AddCommentParams) error
```

Notice the `postID` function parameter corresponding to the `:postID` path parameter. 

I think if you fix that your code should work. If that's not the issue it would be great if you can share a reproducer. 

Regardless, we'll fix the issue in the compiler so the error is reported correctly.

**Comment by saujanreshmi** [comment]
Hi @eandre , Code works fine locally for me, I am only getting this error in my deploys. 

<img width="1025" alt="Screenshot 2023-06-12 at 8 51 21 am" src="https://github.com/encoredev/encore/assets/18434969/22d65212-cfaa-4c06-b0c4-390958c994eb">

**Comment by eandre** [comment]
This is now fixed (#763), thanks very much for reporting!
